httpie/client.py

```diff
@@ -190,10 +191,37 @@ def finalize_headers(headers: RequestHeadersDict) -> RequestHeadersDict:
             if isinstance(value, str):
                 # See <https://github.com/httpie/httpie/issues/212>
                 value = value.encode()
-        final_headers[name] = value
+        final_headers.add(name, value)
     return final_headers
 
 
+def apply_missing_repeated_headers(
+    prepared_request: requests.PreparedRequest,
+    original_headers: RequestHeadersDict
+) -> None:
+    """Update the given `prepared_request`'s headers with the original
+    ones. This allows the requests to be prepared as usual, and then later
+    merged with headers that are specified multiple times."""
+
+    new_headers = RequestHeadersDict(prepared_request.headers)
+    for name, value in prepared_request.headers.items():
+        if name not in original_headers:
+            continue
+
+        original_values = original_headers.getall(name)
+        if value not in original_values:
+            # If the current value is not among the initial values
+            # set for this field, then it means that this field got
+            # overriden on the way, and we should preserve it.
+            continue
+
+        new_headers.popone(name)
+        for value in original_values:
+            new_headers.add(name, value)
+
+    prepared_request.headers = new_headers
```

Comment:
Here we use the name `value` for both the original and prepared values inside the same loop. Let’s prefix the names to reflect the difference and make it easier to follow.

Reply:
`prepared_value`/`original_value` distinction is now added.